### DISCUSSION TITLE
unitful within pluto problems because of modules

### DISCUSSION THREAD

**Discussion by drdozer** [fonsp/Pluto.jl#1374] [Questions about using Pluto] [answered]
I'm trying to use Unitful within pluto to define a unit with the @unit macro, but hat requires the unit to be defined within a module for it to later be imported. I can't find a way to achieve this, as modules defined in one cell aren't visible in another, and cells that define modules can't have any additional code. My units module cell looks like this:

```
module MyUnits
	using Unitful
	@unit tick "τ" Tick (1//60)u"s" false
end
```

I have tried work-arounds like this:

```
begin
	@unit tick "τ" Tick (1//60)u"s" false
	Unitful.register(@__MODULE__)
end 
```

However, this gets upset about tick being defined in multiple workspaces as soon as anything is re-evaluated. I'm guessing this is due to some deeper voodoo in how pluto works interacting bacly with modules?

**Comment by disberd** [comment, marked as answer]
> I'm trying to use Unitful within pluto to define a unit with the @Unit macro, but hat requires the unit to be defined within a module for it to later be imported. **I can't find a way to achieve this, as modules defined in one cell aren't visible in another**

I don't see this as true, I tried creating a new unit in a module and then reference it in another cell and I don't get any problem with that:
![image](https://user-images.githubusercontent.com/12846528/128628545-fc3697a5-5f7e-4792-972e-165141cd84bd.png)

I am on the main branch of Pluto, so don't know if this is somehow working now due the macro analysis feature that is not in the latest registered Pluto version yet, but I think being able to reference created module is there also in base Pluto.

**Reply by drdozer** [reply]
Wow! OK, so this seems to be partially working for me now. 
![image](https://user-images.githubusercontent.com/583516/128685679-02c5142f-a98d-4dc4-88a7-a58ca2f94684.png)

**Reply by drdozer** [reply]
It looks like this is user error on my part.  I was using the wrong arguments to the @unit macro as the quotable unit name. Thanks for the help.
